openapi3/refs.tmpl

```diff
@@ -40,6 +40,14 @@ func (x *{{ $type.Name }}Ref) CollectionName() string { return "{{ $type.Collect
 // RefPath returns the path of the $ref relative to the root document.
 func (x *{{ $type.Name }}Ref) RefPath() *url.URL { return &x.refPath }
 
+func (x *{{ $type.Name }}Ref) setRefPath(u *url.URL) {
+	if x.refPath != (url.URL{}) || u == nil {
```

Comment:
nitpick: let's switch the sides of `||` so the simpler check probably gets called first :)

Reply:
Sure. I'm not actually a fan of this logic anyway but haven't had the time to invest in making it better.